src/domain/notifications/index.ts

```diff
@@ -8,3 +10,137 @@ export const NotificationType = {
   OnchainPayment: "onchain_payment",
   LnInvoicePaid: "paid-invoice",
 } as const
+
+export const NotificationChannel = {
+  Push: "push",
+} as const
+
+export const GaloyNotificationCategories = {
+  Payments: "Payments" as NotificationCategory,
+  Balance: "Balance" as NotificationCategory,
+  AdminPushNotification: "AdminPushNotification" as NotificationCategory,
+} as const
+
+export const checkedToNotificationCategory = (
+  type: string,
+): NotificationCategory | ValidationError => {
+  // TODO: add validation
+  if (!type) {
+    return new InvalidNotificationSettingsError("Invalid notification category")
+  }
+
+  return type as NotificationCategory
+}
+
+export const setNotificationChannelIsEnabled = ({
+  notificationSettings,
+  notificationChannel,
+  enabled,
+}: {
+  notificationSettings: NotificationSettings
+  notificationChannel: NotificationChannel
+  enabled: boolean
+}): NotificationSettings => {
+  const notificationChannelSettings = notificationSettings[notificationChannel]
+  const enabledChanged = notificationChannelSettings.enabled !== enabled
+
+  const newNotificationSettings = {
+    enabled,
+    disabledCategories: enabledChanged
+      ? []
+      : notificationChannelSettings.disabledCategories,
+  }
+
+  return {
+    ...notificationSettings,
+    [notificationChannel]: newNotificationSettings,
+  }
+}
+
+export const enableNotificationCategory = ({
+  notificationSettings,
+  notificationChannel,
+  notificationCategory,
+}: {
+  notificationSettings: NotificationSettings
+  notificationChannel?: NotificationChannel
+  notificationCategory: NotificationCategory
+}): NotificationSettings => {
+  const notificationChannelsToUpdate: NotificationChannel[] = notificationChannel
+    ? [notificationChannel]
+    : Object.values(NotificationChannel)
+
+  let newNotificationSettings = notificationSettings
+
+  for (const notificationChannel of notificationChannelsToUpdate) {
+    const notificationChannelSettings = notificationSettings[notificationChannel]
+    const disabledCategories = notificationChannelSettings.disabledCategories
+
+    const newNotificationChannelSettings = {
+      enabled: notificationChannelSettings.enabled,
+      disabledCategories: disabledCategories.filter(
+        (category) => category !== notificationCategory,
+      ),
+    }
+
+    newNotificationSettings = {
+      ...notificationSettings,
+      [notificationChannel]: newNotificationChannelSettings,
+    }
+  }
+
+  return newNotificationSettings
+}
+
+export const disableNotificationCategory = ({
+  notificationSettings,
+  notificationChannel,
+  notificationCategory,
+}: {
+  notificationSettings: NotificationSettings
+  notificationChannel?: NotificationChannel
+  notificationCategory: NotificationCategory
+}): NotificationSettings => {
+  const notificationChannelsToUpdate: NotificationChannel[] = notificationChannel
+    ? [notificationChannel]
+    : Object.values(NotificationChannel)
```

Comment:
what is the purpose of this?

Reply:
This is the behavior that you requested where it will update all channels if a specific channel in not set.